integrations/anthropic/src/haystack_integrations/components/generators/anthropic/chat/chat_generator.py

```diff
@@ -119,13 +131,40 @@ def _convert_messages_to_anthropic_format(
             i += 1
             continue
 
-        content: List[Union[TextBlockParam, ToolUseBlockParam, ToolResultBlockParam]] = []
-
-        if message.texts and message.texts[0]:
-            text_block = TextBlockParam(type="text", text=message.texts[0])
-            if cache_control:
-                text_block["cache_control"] = cache_control
-            content.append(text_block)
+        content: List[Union[TextBlockParam, ToolUseBlockParam, ToolResultBlockParam, ImageBlockParam]] = []
+
+        # Handle multimodal content (text and images) preserving order
+        for part in message._content:
+            if isinstance(part, TextContent) and part.text:
+                text_block = TextBlockParam(type="text", text=part.text)
+                if cache_control:
+                    text_block["cache_control"] = cache_control
+                content.append(text_block)
+            elif isinstance(part, ImageContent):
+                if not message.is_from(ChatRole.USER):
+                    msg = "Image content is only supported for user messages"
+                    raise ValueError(msg)
+
+                # Anthropic vision API format
+                media_type = part.mime_type or "image/jpeg"
+                if media_type not in IMAGE_SUPPORTED_FORMATS:
+                    supported_formats = ", ".join(IMAGE_SUPPORTED_FORMATS)
+                    msg = (
+                        f"Unsupported image format: {media_type}. "
+                        f"Anthropic supports the following formats: {supported_formats}"
+                    )
+                    raise ValueError(msg)
+                image_block = ImageBlockParam(
+                    type="image",
+                    source={
+                        "type": "base64",
+                        "media_type": cast(Literal["image/jpeg", "image/png", "image/gif", "image/webp"], media_type),
```

Comment:
Since we are now checking the mime types, is `cast` still needed to prevent type errors?